docker-app/qfieldcloud/core/views/files_views.py

```diff
@@ -135,25 +139,48 @@ def get(self, request, projectid, filename):
 
     # TODO refactor this function by moving the actual upload and Project model updates to library function outside the view
     def post(self, request, projectid, filename, format=None):
-        project = Project.objects.get(id=projectid)
+        """
+        Handle POST requests.
+        CURRENTLY UNDERGOING DEBUG:
+            - Write traceback to buffer + deepcopy request before anything happens to the request object
+        """
+        buffer = io.StringIO()
+        print_stack(limit=20, file=buffer)
+        interesting_fields = {
+            "data": copy.deepcopy(request.data),
+            "meta": copy.deepcopy(request.META),
+            "file_names": request.FILES.keys(),
+            "post": copy.deepcopy(request.POST),
+        }
 
         if "file" not in request.data:
-            logger.info(
-                'The key "file" was not found in `request.data`.',
-                extra={
-                    "data_for_key_text_content": str(request.data.get("text", ""))[
-                        :1000
-                    ],
-                    "data_for_key_text_len": len(request.data.get("text", "")),
-                    "request_content_length": request.META.get("CONTENT_LENGTH"),
-                    "request_content_type": request.META.get("CONTENT_TYPE"),
-                    "request_data": list(request.data.keys()),
-                    "request_files": list(request.FILES.keys()),
-                },
+            logging.warning(
+                'The key "file" was not found in `request.data`. Sending report to Sentry.'
+            )
+            report_to_sentry(
+                name=f"{request.META.get('X-Request-Id')}_{projectid}",
+                interesting_fields=interesting_fields,
+                buffer=buffer,
             )
             raise exceptions.EmptyContentError()
 
+        if not request.FILES.getlist("file"):
+            message = 'The key "file" occurs in `request.data` but maps to an empty list. Sending report to Sentry.'
+            logging.warning(message)
+            report_to_sentry(
+                name=f"{request.META.get('X-Request-Id')}_{projectid}",
+                filename=filename,
+                interesting_fields=interesting_fields,
+                buffer=buffer,
+            )
+            raise exceptions.EmptyContentError()
+
+        if len(request.FILES.getlist("file")) > 1:
+            raise exceptions.MultipleContentsError()
+
+        project = Project.objects.get(id=projectid)
```

Comment:
I would check if `request.project` would work in this context?

Reply:
> I would check if `request.project` would work in this context?

Does not seem to work: https://github.com/opengisch/qfieldcloud/actions/runs/5056234567/jobs/9073441192#step:10:3735

Reply:
thanks!

Reply:
Well there's a middleground: get the project from the request if it's defined else from the db